python-threatexchange/threatexchange/cli/match_cmd.py

```diff
@@ -196,18 +230,23 @@ def execute(self, settings: CLISettings) -> None:
             for s_type, index in indices:
```

Comment:
Before processing this list of files, if rotations is true here, generate new files that you iterate through.

Here's one way to do that

```
def handle_rotations() -> Iterator[(Path, t.Optional[RotationType)]:
  for file in self.files:
    if not self.rotations:
      yield file, None
      continue
   for rot, dat in PhotoContent.blah():
      with NamedTemporary(...) as f:
         yield Path(f.name), rot

for path in handle_rotations(self.files):
```